fvtest/porttest/omrsockTest.cpp

```diff
@@ -104,14 +105,66 @@ TEST(PortSockTest, per_thread_buffer_functionality)
 	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
 
 	omrsock_addrinfo_t hints = NULL;
+	uint32_t length = 0;
 	int32_t family = 0;
-	int32_t sockType = 1;
-	int32_t protocol = 1;
-	int32_t flags = 0;
+	int32_t sockType = 0;
+	int32_t protocol = 0;
 
-	OMRPORTLIB->sock_getaddrinfo_create_hints(OMRPORTLIB, &hints, family, sockType, protocol, flags);
+	int32_t hintsFamily = 0;
+	int32_t hintsSockType = 1;
+	int32_t hintsProtocol = 1;
+	int32_t hintsFlags = 0;
+
+	OMRPORTLIB->sock_getaddrinfo_create_hints(OMRPORTLIB, &hints, hintsFamily, hintsSockType, hintsProtocol, hintsFlags);
 	
 	ASSERT_NE(hints, (void *)NULL);
+
+	/* Testing invalid arguments: Pointer to OMRAddrInfoNode that is NULL */
+
+	int32_t rc;
+	rc = OMRPORTLIB->sock_getaddrinfo_length(OMRPORTLIB, NULL, &length);
+	EXPECT_EQ(rc, OMRPORT_ERROR_INVALID_ARGUMENTS);
+
+	rc = OMRPORTLIB->sock_getaddrinfo_family(OMRPORTLIB, NULL, &family, 0);
+	EXPECT_EQ(rc, OMRPORT_ERROR_INVALID_ARGUMENTS);
+
+	rc = OMRPORTLIB->sock_getaddrinfo_socktype(OMRPORTLIB, NULL, &sockType, 0);
+	EXPECT_EQ(rc, OMRPORT_ERROR_INVALID_ARGUMENTS);
+
+	rc = OMRPORTLIB->sock_getaddrinfo_protocol(OMRPORTLIB, NULL, &protocol, 0);
+	EXPECT_EQ(rc, OMRPORT_ERROR_INVALID_ARGUMENTS);
+
+	/* Testing invalid arguments: Index is bigger than the length when querying */
+
+	rc = OMRPORTLIB->sock_getaddrinfo_length(OMRPORTLIB, hints, &length);
+	EXPECT_EQ(rc, 0);
+
+	rc = OMRPORTLIB->sock_getaddrinfo_family(OMRPORTLIB, hints, &family, length + 1);
```

Comment:
If I understand correctly, the addrinfo list is zero indexed. That means that, if you have three elements, their indices are [0, 1, 2]. The length of the chain, 3, is already "one past the end". There is no need to use length + 1.

You should also test for negative indices.

Reply:
You're right, sounds good, I will fix that. Thank you Robert!